### ISSUE TITLE
Help with KM200 setup/credentials

### ISSUE THREAD

**Issue by ansorg** [bosch-thermostat/home-assistant-bosch-custom-component#66]
hello, I try to use the component with a KM200. It doesn't work 
`homeassistant    | 2021-04-16 07:12:09 ERROR (MainThread) [bosch_thermostat_client.encryption.base_encryption] Unable to decrypt: b"y>^\xe9\xb6s\xe2+\xda\xb4\xf4\x8e\x89\x0f\x0e\x1b d\xddiA\xb7x\x7f\xb5+q4\xffFLv\x93\xc6\xfbSG\x83h\x15\x980\xb2|\xb8\xd7z\x88\x1c>&\xefm\xa0\xac%@G\xc0Q\xd8\xac1\x9a\xb7,5\xe8\x83\r\xe8\xf1\xaeDIG\x9d!\xc7\\\x1c\xfb\xc7\x8f\x93\x98\xb1[6\xbby\xa15\x8b\x15Xe\x81Z9D\xcf\x8b\xcd8\xac\x8c G\x94\xe9(j\x95E\xcb\xb9'.\xb2a\x00\x99|4\xf0\x1a\x07" with error: 'utf-8' codec can't decode bytes in position 3-4: invalid continuation byte`

probably because I don't understand the instructions. 

>password: (string)(Required) password to your gateway which you created.
>access_token: (string)(Required) Access token from sticker at your gateway.

On the KM200 I have no "access_token" but
- a "Loginname" which is 9 numerics
- a "Passwort" in the form `xxxx-xxxx-xxxx-xxxx `
- an undefined thing, a serial number in the form `xxxx-xxxxxxxxx-xxxxxxxxxx`
- MAC address

The device **is** connected with www.buderus-connect.de and there I use a password that I set when the technician installed the KM200 and did the initial setup

Which item do I need to use for the login in which place?

**Comment by pszafer** [comment]
Did you follow this wiki -> https://github.com/bosch-thermostat/home-assistant-bosch-custom-component/wiki/Integrations-config ?

**Comment by ansorg** [comment]
ah thanks! That page is clear on what to use. 

Unfortunately, I still get a similar error. Any idea what to do with that?
Something wrong with my device? (as noted above, it is connected with buderus online and that seems to work fine)

` ERROR (MainThread) [bosch_thermostat_client.encryption.base_encryption] Unable to decrypt: b"y>^\xe9\xb6s\xe2+\xda\xb4\xf4\x8e\x89\x0f\x0e\x1b d\xddiA\xb7x\x7f\xb5+q4\xffFLv\x93\xc6\xfbSG\x83h\x15\x980\xb2|\xb8\xd7z\x88\x1c>&\xefm\xa0\xac%@G\xc0Q\xd8\xac1\x9a\xb7,5\xe8\x83\r\xe8\xf1\xaeDIG\x9d!\xc7\\\x1c\xfb\xc7\x8f\x93\x98\xb1[6\xbby\xa15\x8b\x15Xe\x81Z9D\xcf\x8b\xcd8\xac\x8c G\x94\xe9(j\x95E\xcb\xb9'.\xb2a\x00\x99|4\xf0\x1a\x07" with error: 'utf-8' codec can't decode bytes in position 3-4: invalid continuation byte`

**Comment by pszafer** [comment]
`www.buderus-connect.de` will be shutdown in a month, can you connect using Buderus MyDevice app?

**Comment by ansorg** [comment]
I use that app since the device got installed. Although, it only works "online" but not with the option "local connection only" (roughly translated from german)

**Comment by pszafer** [comment]
it should work then with both version.
choose:
- first screen: IVT
- HTTP
- put IP address of gateway,
- put ACCESS TOKEN from sticker without `-`
- put PASSWORD you use to login to the app

**Comment by ansorg** [comment]
hi Pawel, thanks for listening

It should, but it doesn't. And the full story sounds totally weird, to me at least.

The KM200 got installed by a technican about a year ago, including the connection to the online service. At some point I just had to set (or change?) my password for the app/website. 
I had OpenHAB running already and added a Binding for the KM200 and it worked (using a website that would generate a token based on my device/account credentials)

One year later I needed to change the network setup. KM200 now via Ethernet on the router instead through a WiFi repeater as before, got new IP address. Therefore OpenHAB Binding not working anymore. I tried to reconnect that and it failed with encryption errors. I asked for support/ideas and noting so far. 

So I went to hass.io and this component. And here I run into the same issues with decrypting the communication from the local device, sigh. 

So, something seems wrong with the credentials that I use to connect. But why only after the network change?? 

And what could be wrong? The data from the sticker is the same ... my Password from the app? But I can still log into the app/website with the same password that I try to use for the OHAB/HASS connection. How could that not work? I don't get it. 

Just looking for any ideas what else to try/how to debug

**Comment by pszafer** [comment]
if you want you can PM me on Discord and I can check if I can fetch data with your credentials. Then you'd know if this is a network bug or your environment is the reason

**Comment by pszafer** [comment]
@ansorg are you trying to connect to XMPP or HTTP endpoint?

**Comment by ansorg** [comment]
HTTP in the local network

**Comment by pszafer** [comment]
try XMPP, maybe you have some firewall between HA and your gateway.

**Comment by ansorg** [comment]
Fixed it! 

It couldn't be the network and your comments made me think and search again ... 

I found the manual for the KM200 and in there a "reset device" instruction. Ok, brute-force approach? 
In the end I didn't do a full reset but somehow managed to reset the user password by re-registering the gateway in the mobile app! 

And with this fresh new password the device registration in the hass.io component (and also in OpenHAB) works and I have all my data available again. 

Thanks for listening and providing ideas.
